src/js/components/Calendar/__tests__/Calendar-test.js

```diff
@@ -146,7 +161,288 @@ describe('Calendar', () => {
     );
     expect(container.firstChild).toMatchSnapshot();
     fireEvent.click(getByText('17'));
-    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2018-01-17T/));
+    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2020-01-17T/));
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('first day sunday week monday', () => {
+    // When the first day of the month is Sunday,
+    // and the request of firstDayOfWeek
+    // is Monday, we are verifing we are not missing a week, issue 3253.
+    const { container } = render(
+      <Grommet>
+        <Calendar
+          firstDayOfWeek={1}
+          date="2020-03-01T00:00:00-08:00"
+          animate={false}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('change months', () => {
+    jest.useFakeTimers();
+    const { container, getByLabelText } = render(
+      <Grommet>
+        <Calendar date={DATE} />
+      </Grommet>,
+    );
+    // Change the Calendar from January to December
+    fireEvent.click(getByLabelText('December 2019'));
+    act(() => {
+      jest.advanceTimersByTime(400);
```

Comment:
This seems a bit brittle in that the value is intended to align with the value in the code. Can we just use `jest.runAllTimers();`?

Reply:
Yes, I changed the test to use `jest.runAllTimers()` instead of `jest.advanceTimersByTime(400);`